framework/src/base/NonlinearEigenSystem.C

```diff
@@ -152,4 +154,20 @@ NonlinearEigenSystem::addEigenKernels(MooseSharedPointer<KernelBase> kernel, THR
     _non_eigen_kernels.addObject(kernel, tid);
 }
 
+void
+NonlinearEigenSystem::checkEigenBoundaryConditionConsistency(MooseSharedPointer<BoundaryCondition> bc, InputParameters & parameters)
+{
+  MooseSharedPointer<IntegratedBC> ibc = MooseSharedNamespace::dynamic_pointer_cast<IntegratedBC>(bc);
+  if (ibc)
+    mooseError("Can't set an inhomogeneous integrated boundary for eigenvalue problems.");
+
+  MooseSharedPointer<DirichletBC> nbc = MooseSharedNamespace::dynamic_pointer_cast<DirichletBC>(bc);
+  if (nbc && parameters.get<Real>("value"))
+    mooseError("Can't set an inhomogeneous Dirichlet boundary for eigenvalue problems.");
+
+  if (!nbc)
+    mooseError("Invalid NodalBC for eigenvalue problems, please use homogeneous Dirichlet.");
+}
```

Comment:
You want to say "boundary condition" in these error messages, "boundary" means the physical "edge" of the mesh.